app/components/Settings/Preview/chart-editor.jsx

```diff
@@ -0,0 +1,137 @@
+     * @param {Array}         [props.plotlyJSON.data] - Plotly graph data
+     * @param {Array}         [props.plotlyJSON.frames] - Plotly graph frames
+     * @param {object}        [props.plotlyJSON.layout] - Plotly graph layout
+     * @param {function}      props.onUpdate - Callback invoked to update plotlyJSON
+     *
+     * @param {hidden}        props.hidden - If hidden, don't mount <PlotEditor>
+     */
+    constructor(props) {
+        super(props);
+
+        /**
+         * @member {object} state - Component state
+         * @property {object} state.dataSources - A data source per column (used by <PlotlyEditor>)
+         *
+         * @property {Array.<object>} state.dataSourceOptions - Data source options (used by <PlotlyEditor>)
+         * @property {string} state.dataSourceOptions.label - Data source label (used by <PlotlyEditor>)
+         * @property {string} state.dataSourceOptions.value - Data source index in dataSources (used by <PlotlyEditor>)
+         */
+        this.state = ChartEditor.computeDataSources(this.props);
+    }
+
+    static computeDataSources(props) {
+        const {columnNames, rows} = props;
+
+        const dataSources = {};
+        columnNames.forEach(name => {
+            dataSources[name] = [];
+        });
+
+        // Cap plots to 100k rows
+        const length = Math.min(rows.length, 100000);
+
+        for (let i = 0; i < length; i++) {
+            const row = rows[i];
+
+            for (let j = 0; j < row.length; j++) {
+                const value = row[j];
+                const columnName = columnNames[j];
+                dataSources[columnName].push(value);
+            }
+        }
+
+        const dataSourceOptions = columnNames.map(name => ({value: name, label: name}));
+
+        return {dataSources, dataSourceOptions};
+    }
+
+    componentWillReceiveProps(nextProps) {
+        // Did props change?
+        const {columnNames, rows} = this.props;
+        const {columnNames: nextColumnNames, rows: nextRows} = nextProps;
+
+        let isEqual = (columnNames.length === nextColumnNames.length);
+        for (let i = 0; isEqual && (i < columnNames.length); i++) {
+            isEqual = isEqual && (columnNames[i] === nextColumnNames[i]);
+            isEqual = isEqual && (rows[i] === nextRows[i]);
+        }
+
+        // If it did, update the data sources
+        if (!isEqual) {
+            this.setState(ChartEditor.computeDataSources(nextProps));
+        }
+    }
+
+    render() {
+        const {plotlyJSON, onUpdate, hidden} = this.props;
+        const {dataSources, dataSourceOptions} = this.state;
+
+        const config = {
+            editable: true
+        };
+        const data = plotlyJSON && plotlyJSON.data;
+        const frames = plotlyJSON && plotlyJSON.frames;
+        const layout = plotlyJSON && plotlyJSON.layout;
+
+        return (
+            <div
+                ref={'container'}
+                style={{
+                    minHeight: 400,
```

Comment:
Just a suggestion.  Maybe a constant or move to a separate file.  I see the min height of 400 defined in a bunch of files.  Maybe we should centralize to a single file?  your thoughts?

Reply:
I've re-enabled [css-loader](https://github.com/webpack-contrib/css-loader), so that we can import again `.css` files into a component. But I didn't create a `chart-editor.css`, because I feel we need tidy up the CSS in Falcon first:
- perhaps define a `.css` file with a common theme
- follow [Plotly's style guide](https://brand.plot.ly/)
- Jack mentioned  [tachyons](http://tachyons.io/) and I'm keen to give it a go.